Topics/Development_Process/Github_Pages.md

```diff
@@ -0,0 +1,91 @@
+
+1. Create a new repo.
+2. Add your HTML/CSS/js files to your repository.
+   1. You should name your HTML file `index.html` -- that's the "home" page.
+   2. Other than that, if your css/javascript works on your machine if you were to clone this repo, it should also work on GitHub pages.
+3. Go to your repository settings.
+4. Navigate to Pages in the repository's settings.
+5. Choose a branch you want to deploy your page to (it should say `None` by default`). **The moment you select your branch and hit save, GitHub Pages will be enabled and will start deploying.** It should take between 1-2 minutes. This point onwards, your GitHub page will update every time you commit something to the repository.
+
+Your site by default will have this URL:
+
+```
+<your_GitHub_username>.github.io/<repository_name>/
+```
+
+By default, `index.html` will show up.
+
+To access another file in your repository, your URL would be:
+
+```
+<your_GitHub_username>.github.io/<repository_name>/<path_to_file>
+```
+
+For example, if my GitHub username was `user`, my repo name was `repo`, and my repository had these files:
+
+```
+somerandomfolder/a.html
+contact.html
+index.html
+styles.css
+```
+
+Then the links to these files would be, respectively:
+
+```
+https://user.github.io/repo/somerandomfolder/a.html
+https://user.github.io/repo/somerandomfolder/contact.html
+https://user.github.io/repo/somerandomfolder or user.github.io/repo/somerandomfolder/index.html
+https://user.github.io/repo/somerandomfolder/styles.css
+```
+
+## Deploying a React project
+
+As simple as [following this tutorial](https://create-react-app.dev/docs/deployment#github-pages), from create-react-app, a well-known tool for, well, creating a React app.
+
+If you want routing, follow [this](https://www.freecodecamp.org/news/deploy-a-react-app-to-github-pages/). This involves making changes to your React project, so don't worry about touching GitHub for this.
+
+Maybe you might want to [automate the process with GitHub actions?](https://github.com/marketplace/actions/deploy-react-to-github-pages)
+
+**NOTE:** Please don't use this for Next.js projects; Vercel's already got you.
+
+## Deploying an Angular Project
+
+- [Here's your React equivalent.](https://github.com/angular-schule/angular-cli-ghpages)
+
+- [Want to automate this?](https://github.com/marketplace/actions/angular-deploy-gh-pages-actions)
+
+## Will I ever run out of bandwidth?
+
+There's a 100GB monthly *soft* bandwidth limit. That's the amount of data that can be transferred from GitHub to users accessing the site until GitHub starts getting a bit upset.
+
+Ensure the files you transmit are small, because on each page load assuming no caching, the amount of bandwidth you transfer is the sum of the size of all files served to the user that came from your repository (quite unfortunate you can't use Discord as a CDN anymore).
```

Comment:
I think mentioning Discord here is a bit distracting to the reader. If you want to explain more about Discord I think it would be worth leaving in, otherwise I might delete it. If you think it makes sense to just leave this, feel free through.